Leave project and date

ErikSin opened the pull request:
closes #660 
closes #588 

Creates screen indicating how the user can leave a project.
Add the date added to device card.

<img width="283" alt="image" src="https://github.com/user-attachments/assets/0cd6f368-dd4d-4d78-bcd8-557f99faaf3b">

<img width="288" alt="image" src="https://github.com/user-attachments/assets/920048fc-6021-4001-ac55-978f8a2374be">

ErikSin committed:
chore: update device card and device name

ErikSin committed:
chore: translations

achou11 commented on src/frontend/screens/HowToLeaveProject.tsx:
```diff
@@ -0,0 +1,105 @@
+  instructions: {
+    id: 'screens.HowToLeaveProject.instructions',
+    defaultMessage:
+      'To leave this project please uninstall and reinstall Mapeo. All project data will be removed from this device.',
```
```suggestion
      'To leave this project please uninstall and reinstall CoMapeo. All project data will be removed from this device.',
```

ErikSin replied:
https://github.com/digidem/comapeo-mobile/pull/674/commits/df6ed581418aa143f310f33b5335da3ecb0d57d7

ErikSin committed:
chore: update device card

ErikSin commented on src/frontend/screens/Settings/ProjectSettings/YourTeam/index.tsx:
```diff
@@ -173,10 +174,8 @@ export const YourTeam: NativeNavigationComponent<'YourTeam'> = ({
-            deviceInfo.data && deviceInfo.data.name === participant.name
-          }
+          dateAdded={participant.joinedAt}
+          thisDevice={deviceInfo.data?.deviceId === participant.deviceId}
```
this is semi out of scope, but small enough fix that I thought it wouldn't hurt to fix it here

achou11 replied:
also out of scope, but ideally this would make sure that the `deviceInfo` query is in a success state before allowing this check to happen. Otherwise you can expose UI or features (or in a really unfortunate scenario, effects) unintentionally for a brief moment.

Think the work to properly account for that that can happen in an eventual follow-up though

achou11 commented on src/frontend/sharedComponents/DeviceCard.tsx:
```diff
@@ -44,30 +54,60 @@ export const DeviceCard = ({
-        deviceId={deviceId}
-        iconSize={75}
-      />
+      style={[style]}>
```
nit: no need for the array now

```suggestion
      style={style}>
```

ErikSin replied:
https://github.com/digidem/comapeo-mobile/pull/674/commits/da4aa3596be535cd638ec9a58cb90a086605bbb0

achou11 commented:
Before I do a review, does this intentionally overlap with https://github.com/digidem/comapeo-mobile/pull/636?

ErikSin commented:
yes, I spoke to Cindy on Slack about it. There were a few implementation differences that needed to be addressed, and it was blocking the leave project works. Cindy is also away this week, so I let her know that it was easier if I just redid it!

achou11 commented on src/frontend/sharedComponents/DeviceCard.tsx:
```diff
@@ -44,30 +54,60 @@ export const DeviceCard = ({
+              onPress={() => {
+                navigate('HowToLeaveProject');
+              }}>
+              <Text style={{fontSize: 12, color: RED}}>
```
based on what i'm seeing with default OS font settings, should:

- increase font weight
- maybe increase font size?
- maybe use `MAGENTA` instead of `RED`?

Currently:

<img width="200" alt="image" src="https://github.com/user-attachments/assets/aa617fad-42b1-49e1-97e9-78c1cdbf51ce">

Design:

<img width="200" alt="image" src="https://github.com/user-attachments/assets/294ab415-f1af-426d-b5e6-15b5345913d1">

ErikSin replied:
https://github.com/digidem/comapeo-mobile/pull/674/commits/7704b2a2aa999a780a5aa0c368dc5b197b353e89

achou11 commented on src/frontend/sharedComponents/DeviceCard.tsx:
```diff
@@ -44,30 +54,60 @@ export const DeviceCard = ({
+        />
+        <View style={{alignSelf: 'flex-start'}}>
+          {dateAdded && (
+            <Text style={{fontSize: 12, marginBottom: 10}}>
```
based on design, this should be a lighter grey color:

<img width="200" alt="image" src="https://github.com/user-attachments/assets/ed0487c4-d07e-497f-8b1b-c3980edcf758">

ErikSin replied:
https://github.com/digidem/comapeo-mobile/pull/674/commits/7704b2a2aa999a780a5aa0c368dc5b197b353e89

achou11 commented on src/frontend/screens/HowToLeaveProject.tsx:
Might want to check how this looks with default OS font settings. Text basically looks unstyled on the devices I tested:

<img width="200" alt="image" src="https://github.com/user-attachments/assets/4cfc367e-5246-48bf-b36c-f195320a8b03">

<img width="200" alt="image" src="https://github.com/user-attachments/assets/9d78acb4-2386-4fca-aeb2-73b833ae169e">

ErikSin replied:
What styling are you referring to here? Aka what do you think is missing in the default OS settings?

achou11 replied:
I'm referring to the styling of the text (mostly the title text) in the content area on this screen, as seen in the [design](https://www.figma.com/design/uzynHXR0xgJu19lsCoJyXN/%F0%9F%90%9A-Shell-App?node-id=1197-17446&t=77wlSzzaRQIHNBV0-4):

<img width="200" alt="image" src="https://github.com/user-attachments/assets/412d5897-0829-4857-9947-aa5fab0c3fe0">

Compared to what I see on my devices (which use the default OS font size settings), there's a noticeable difference

ErikSin replied:
https://github.com/digidem/comapeo-mobile/pull/674/commits/640afe25a9fe180e964e2a9c8b8c30f466501c39
https://github.com/digidem/comapeo-mobile/pull/674/commits/ab5ebb43ccf041fedfcc4f3b2f8e1654919dc4ec

achou11 commented on src/frontend/screens/HowToLeaveProject.tsx:
```diff
@@ -0,0 +1,105 @@
+          {formatMessage(m.goBack)}
+        </Button>
+      }>
+      <Text style={[styles.text, {fontSize: 20}]}>
```
Need to add the question mark image above this

<img width="200" alt="image" src="https://github.com/user-attachments/assets/a76b47f1-b37f-4658-a753-b694a63f7487">

ErikSin replied:
I think the question mark indicates that a icon is suppose to be chosen, but hasn't yet, but I will confirm with design!

ErikSin replied:
I was wrong, its intentional! I am adding it now

ErikSin replied:
https://github.com/digidem/comapeo-mobile/pull/674/commits/e5ef98f2c37f255ac3e0d35bb30b6d44a5ea5920

achou11 commented on src/frontend/sharedComponents/DeviceCard.tsx:
```diff
@@ -44,30 +54,60 @@ export const DeviceCard = ({
+          {dateAdded && (
+            <Text style={{fontSize: 12, marginBottom: 10}}>
+              <FormattedDate
+                value={new Date(dateAdded)}
```
I don't think this needs to be converted into a date first

```suggestion
                value={dateAdded}
```

doing so doesn't play nicely with React either because you're creating a new instance on each render

ErikSin replied:
https://github.com/digidem/comapeo-mobile/pull/674/commits/7704b2a2aa999a780a5aa0c368dc5b197b353e89

ErikSin committed:
chore: mapeo => CoMapeo

ErikSin committed:
chore: unnecessary array

ErikSin committed:
chore: update deviceCard

ErikSin committed:
chore: title sizing

ErikSin committed:
chore: bold

ErikSin committed:
chore: add question icon

ErikSin commented:
I have updated the styling and added the icon:
<img width="291" alt="image" src="https://github.com/user-attachments/assets/45674921-e407-40fe-a814-9480c5e7e6d3">

ErikSin commented on src/frontend/sharedComponents/icons/QuestionMarkWithShadow.tsx:
I made this its own icon because it needs to be used in another screen, which i will do as a follow up pr

ErikSin committed:
chore: remove unusedimport

ErikSin committed:
chore: remove unneccesary justify content

achou11 commented on src/frontend/screens/HowToLeaveProject.tsx:
```diff
@@ -0,0 +1,115 @@
+  const membersQuery = useProjectMembers();
+  const deviceInfo = useDeviceInfo();
+
+  const coordinators = !membersQuery.data
+    ? []
+    : membersQuery.data.filter(
+        member =>
+          member.role.roleId === COORDINATOR_ROLE_ID ||
+          member.role.roleId === CREATOR_ROLE_ID,
+      );
```
nit: remove the negation

```suggestion
  const coordinators = membersQuery.data
    ? membersQuery.data.filter(
        member =>
          member.role.roleId === COORDINATOR_ROLE_ID ||
          member.role.roleId === CREATOR_ROLE_ID,
      );
    : []
```
